app/uk/gov/hmrc/emcstfe/models/request/SubmitReportOfReceiptRequest.scala

```diff
@@ -31,23 +29,11 @@ case class SubmitReportOfReceiptRequest(body: SubmitReportOfReceiptModel, useFS4
                                        (implicit request: UserRequest[_]) extends ChrisRequest with SoapEnvelope with EisSubmissionRequest with EisMessage {
 
   private val arcCountryCode = body.arc.substring(2, 4)
-  private val traderModelCountryCode: Option[TraderModel] => String = _.flatMap(_.countryCode).getOrElse(Constants.GB)
+  private val ernCountryCode = exciseRegistrationNumber.substring(0, 2)
   private val messageNumber = 818
 
-  val messageRecipient =
-    Constants.NDEA ++ (if (body.destinationType.contains(DirectDelivery)) {
-      traderModelCountryCode(body.consigneeTrader)
-    } else {
-      arcCountryCode
-    })
-
-  val messageSender =
-    Constants.NDEA ++ (body.destinationType match {
-      case Some(TaxWarehouse) => traderModelCountryCode(body.deliveryPlaceTrader)
-      case Some(TemporaryRegisteredConsignee) | Some(RegisteredConsignee) => traderModelCountryCode(body.consigneeTrader)
-      case Some(DirectDelivery) => arcCountryCode
-      case _ => Constants.GB
-    })
+  val messageRecipient = Constants.NDEA ++ arcCountryCode
+  val messageSender: String = Constants.NDEA ++ ernCountryCode
```

Comment:
Would it make sense to change it to this:

**Message Recipient**

- Use ARC

**Message Sender:**

- Tax Warehouse still uses the `DeliveryPlace` trader model
- All others use the `Consignee` trader model
- If it can't find an option then it defaults to the logged in user ERN

This would keep it closer to the original requirement, which does specifically mention `DeliveryPlace` trader in the JIRA ticket for Tax Warehouse: https://jira.tools.tax.service.gov.uk/browse/PAM-2211

And I worry we might break something different if we change the way that is working. As the only problems have been raised with `DirectDelivery` so we just need to fix that one.

i.e.

```suggestion
  private val arcCountryCode = body.arc.substring(2, 4)
  private val traderModelCountryCode: Option[TraderModel] => String = _.flatMap(_.countryCode).getOrElse(exciseRegistrationNumber.substring(0, 2))

  private val messageNumber = 818

  val messageRecipient = Constants.NDEA ++ arcCountryCode

  val messageSender =
    Constants.NDEA ++ (body.destinationType match {
      case Some(TaxWarehouse) => traderModelCountryCode(body.deliveryPlaceTrader)
      case _ => traderModelCountryCode(body.consigneeTrader)
    })
```

Reply:
The other reason for this, is that I _think_ you can submit a Report of Receipt as the `Consignor` to say that the `Consignee` received it.

I'm able to see the `Report of Receipt` message in the inbox in QA as the Consignor and there's a link then to Submit the Report of Receipt. Which let's me access RoR and start to submit. _(If you're not allowed to do that then we have a bug)_

If it is valid to do that then, it might want to still use the `Consignee` trader model for the Message Sender; even though the RoR is being submitted by the Consignor.

I don't know what would happen if the Draft Movement was sent to say "NDEA.FR" as the Message Recipient (Consignee). And then the RoR wasn't returned with "NDEA.FR" as the Message Sender (Consignee). If it had originated from say "GB" and we used the logged in Consignor ERN as the Message Sender for the RoR it wouldn't match the fact it was actually sent to "FR".

Reply:
Updated on https://github.com/hmrc/emcs-tfe/pull/156/commits/91f1c9828dac914a8ea652daf956bb74b788ba76